src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -28,39 +28,46 @@ public NotificationService(NotificationRepository notificationRepository, Verify
 
     public void acceptNotification(long notificationId) {
 
-        checkNotificationExists(notificationId);
         Notification notification = notificationRepository.findByNotificationId(notificationId);
-        checkWrongReceiver(notification);
-
-        switch (notification.getType()) {
-        case MESSAGE:
-            deleteNotification(notification);
-            break;
-        case MATCH_VERIFICATION:
-            verifyMatchService.acceptRequest((MatchVerificationRequest) notification);
-            break;
+        if (notification == null) {
```

Comment:
Why are you doing the null check here instead of in `checkNotificationExists()`?
Anyway there's a nicer way to do this. The CrudRepository provides a method named `findById()` which returns an `Optional` that has the method `orElseThrow(Exception.class)` which either returns the object in the Optional or throws an Exception which is nicer than doing the check by hand. You can do that in your service method for getting a user by id.